shipyard_blueprints/airbyte/shipyard_airbyte/cli/cli_trigger_sync.py

```diff
@@ -27,7 +23,40 @@ def main():
 
     # save sync response as variable
     shipyard.logs.create_pickle_file(artifact_subfolder_paths,
-                                     'sync_response', resp['jobId'])
+                                     'sync_response', job_id)
+
+
+def main():
+    args = get_args()
+    connection_id = args.connection_id
+    api_token = args.api_token
+
+    client = AirbyteClient(access_token=api_token)
+    try:
+        resp = client.trigger_sync(connection_id=connection_id)
+    except Exception as e:
+        client.logger.error(
+            f"Error occurred when attempting to trigger sync due to the following error: {e}")
+        sys.exit(1)
+
+    job_id = resp['jobId']
+
+    generate_pickle_file(job_id)  # This is for backwards compatibility the Check Sync Blueprint needs in order to run
+
+    if args.wait_for_completion == 'TRUE' and (0 < args.poke_interval <= 60):
+        status = client.determine_sync_status(job_id=job_id)
+        while status not in (client.EXIT_CODE_FINAL_STATUS_COMPLETED,
+                             client.EXIT_CODE_FINAL_STATUS_INCOMPLETE,
+                             client.EXIT_CODE_FINAL_STATUS_CANCELLED):
+            status = client.determine_sync_status(job_id=job_id)
+            time.sleep(args.poke_interval)
+
+        sys.exit(status)
+    elif args.wait_for_completion == 'TRUE':
```

Comment:
What if the args.wait_for_completion != 'TRUE'?

Reply:
the script will be completed it should exit normally. It is assumed that if wait for completion is not TRUE then either it is a legacy blueprint or the user just wants to trigger the sync

Reply:
I see now. Couldn't you just check the poke_interval beforehand and exit if it doesn't satisfy criteria? Something like this:

```
if not 0 < args.poke_interval <= 60:
    sys.exit(<exit_code>)

    if args.wait_for_completion == 'TRUE':
        status = client.determine_sync_status(job_id=job_id)
        while status not in (client.EXIT_CODE_FINAL_STATUS_COMPLETED,
                             client.EXIT_CODE_FINAL_STATUS_INCOMPLETE,
                             client.EXIT_CODE_FINAL_STATUS_CANCELLED):
            status = client.determine_sync_status(job_id=job_id)
            time.sleep(args.poke_interval)
```

Reply:
My indents got screwed up, but you get the picture

Reply:
the poke interval should only be taken into consideration if wait for completion is true since that is the only time it is used